include/drivers/video.h

```diff
@@ -0,0 +1,518 @@
+{
+	const struct video_driver_api *api =
+		(const struct video_driver_api *)dev->driver_api;
+
+	return api->stream_start(dev);
+}
+
+/**
+ * @brief Stop the capture or output process.
+ *
+ * @retval 0 Is successful.
+ * @retval -EIO General input / output error.
+ */
+static inline int video_stream_stop(struct device *dev)
+{
+	const struct video_driver_api *api =
+		(const struct video_driver_api *)dev->driver_api;
+
+	return api->stream_stop(dev);
+}
+
+/**
+ * @brief Set the value of a control.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param cid Control ID.
+ * @param value Pointer to the control value.
+ *
+ * @retval 0 Is successful.
+ * @retval -EINVAL If parameters are invalid.
+ * @retval -ENOTSUP If format is not supported.
+ * @retval -EIO General input / output error.
+ */
+static inline int video_set_ctrl(struct device *dev, unsigned int cid,
+				 void *value)
+{
+	const struct video_driver_api *api =
+		(const struct video_driver_api *)dev->driver_api;
+
+	if (api->set_ctrl == NULL) {
+		return -ENOTSUP;
+	}
+
+	return api->set_ctrl(dev, cid, value);
+}
+
+/**
+ * @brief Get the current value of a control.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param cid Control ID.
+ * @param value Pointer to the control value.
+ *
+ * @retval 0 Is successful.
+ * @retval -EINVAL If parameters are invalid.
+ * @retval -ENOTSUP If format is not supported.
+ * @retval -EIO General input / output error.
+ */
+static inline int video_get_ctrl(struct device *dev, unsigned int cid,
+				 void *value)
+{
+	const struct video_driver_api *api =
+		(const struct video_driver_api *)dev->driver_api;
+
+	if (api->get_ctrl == NULL) {
+		return -ENOTSUP;
+	}
+
+	return api->get_ctrl(dev, cid, value);
+}
+
+
+/**
+ * @brief Get the capabilities of a video endpoint.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param ep Endpoint ID.
+ * @param caps Pointer to the video_caps struct to fill.
+ *
+ * @retval 0 Is successful, -ERRNO code otherwise.
+ */
+static inline int video_get_caps(struct device *dev,
+				 enum video_endpoint_id ep,
+				 struct video_caps *caps)
+{
+	const struct video_driver_api *api =
+		(const struct video_driver_api *)dev->driver_api;
+
+	return api->get_caps(dev, ep, caps);
+}
+
+
+/**
+ * @brief Register event callback for a video endpoint.
+ *
+ * Register a endpoint event handler which is called on frame completion,
+ * error, etc (cf video_event enum). There is no guarentee on the
+ * context, callback can be executed in IRQ or Thread context.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param ep Endpoint ID.
+ * @param cb Event callback function.
+ * @param user_data Pointer to user data, passed back event callback
+ *
+ * @retval 0 Is successful, -ERRNO code otherwise.
+ */
+static inline int video_register_event_cb(struct device *dev,
```

Comment:
How I can de-register callback?

Reply:
Either I add the symmetric or consider passing NULL cb to register_function

Reply:
IMHO NULL is better